server/src/testFixtures/java/io/crate/operation/aggregation/AggregationTestCase.java

```diff
@@ -225,12 +225,19 @@ public Object executeAggregation(Signature maybeUnboundSignature,
                 minNodeVersion,
                 optionalParams
             );
+            var resultWithoutDocValues = aggregationFunction.terminatePartial(
+                RAM_ACCOUNTING,
+                partialResultWithoutDocValues
+            );
             // assert that aggregations with/-out doc values yield the
             // same result, if a doc value aggregator exists.
             if (partialResultWithDocValues != null) {
                 assertThat(partialResultWithDocValues).hasSize(1);
-                assertThat(partialResultWithoutDocValues).isEqualTo(
-                    partialResultWithDocValues.get(0).get(0));
+                var resultWithDocValues = aggregationFunction.terminatePartial(
+                    RAM_ACCOUNTING,
+                    partialResultWithDocValues.get(0).get(0)
+                );
+                assertThat(resultWithoutDocValues).isEqualTo(resultWithDocValues);
```

Comment:
We compare now the final results, not the partial results, therefore the various topk-states are independent now.